dispatcher/backend/src/common/schemas/offliners/freecodecamp.py

```diff
@@ -1,54 +1,67 @@
-from marshmallow import fields
+from marshmallow import fields, validate
 
-from common.schemas import SerializableSchema
+from common.schemas import LongString, SerializableSchema, String, StringEnum
 from common.schemas.fields import (
     validate_output,
     validate_zim_description,
     validate_zim_filename,
     validate_zim_longdescription,
 )
 
+FCC_LANG_MAP = {
+    "ara": "arabic",
+    "cmn": "chinese",
+    "lzh": "chinese-traditional",
+    "eng": "english",
+    "spa": "espanol",
+    "deu": "german",
+    "ita": "italian",
+    "jpn": "japanese",
+    "por": "portuguese",
+    "ukr": "ukranian",
+}
+
 
 class FreeCodeCampFlagsSchema(SerializableSchema):
     class Meta:
         ordered = True
 
-    course = fields.String(
+    course = String(
         metadata={
             "label": "Course(s)",
             "description": "Course or course list (separated by commas)",
         },
         required=True,
     )
 
-    language = fields.String(
+    language = StringEnum(
         metadata={
             "label": "Language",
-            "description": "Language of zim file and curriculum. Either (without "
-            "quotes) 'ara' (arabic), 'cmn' (chinese), 'lzh' (chinese-traditional), "
-            "'eng' (english), 'spa' (espanol), 'deu' (german), 'ita' (italian), "
-            "'jpn' (japanese), 'por' (portuguese), 'ukr' (ukranian).",
+            "description": "Language of zim file and curriculum. One of "
```

Comment:
Given the UI will present this as a choice field, maybe we can spare this… but it gives the corresponding label so I don't know

Reply:
The idea is indeed to present the corresponding label, especially since we support both `cmn` and `lzh` which are far than obvious for most users. 
Should the list of supported languages grow bigger, I agree that we will have to get rid of this (but I hope that at that future time, we will to be able to present the label in the combobox and associate it with the proper key, transparently).

Reply:
Just opened https://github.com/openzim/zimfarm/issues/836